### DISCUSSION TITLE
Separating `@trpc/client` from `@trpc/server` completely

### DISCUSSION THREAD

**Discussion by iam-kevin** [trpc/trpc#4242] [Q&A]
Currently, it seems that to use `@trpc/client`, I also need to have `@trpc/server` installed to not get any funny behaviors. I have installed these parts in different projects of a mono repo, and wanting to know if we can have that not happen. Maybe I can contribute to that?

**Comment by Nick-Lucas** [comment]
There is the possibility of having an `@trpc/core` package with the shared stuff, we've discussed it briefly quite recently as I had the same idea. Actual benefits though aren't so clear, other than minimising possible confusion why something called `server` is in the client.

**Comment by iam-kevin** [comment]
Could the reasoning be that this makes it possible for other types of extension be possible without bloating the package(s) i.e. `@trpc/core`, `@trpc/client`, `@trpc/server`... then have these different parts be built be used by the community for their own types of use cases
-  `react-native-trpc-client` (currently frankensteining my own, but it would be great if there was a separation)
- `solid-expo-client`
- ....

I understand that the power of tRPC is an environment of shared codebase, but with Monorepos and Monorepo managers (the likes of Turborepo and Nx) being a big deal now. It might be something to consider
